lib/foreman_maintain/package_manager/dnf.rb

```diff
@@ -1,10 +1,44 @@
 module ForemanMaintain::PackageManager
   class Dnf < Yum
+    PROTECTOR_CONFIG_FILE = '/etc/yum/pluginconf.d/foreman-protector.conf'.freeze
+    PROTECTOR_WHITELIST_FILE = '/etc/yum/pluginconf.d/foreman-protector.whitelist'.freeze
+
     def clean_cache(assumeyes: false)
       dnf_action('clean', 'all', :assumeyes => assumeyes)
       super
     end
 
+    def dnf_plugin_dirs
+      dnf_plugin_parent_dirs = Dir.glob('/usr/lib/python3*').map { |d| d + '/site-packages' }
+      dnf_plugin_parent_dirs.map do |d|
+        if Dir.new(d).children.include?('dnf-plugins')
+          d + '/dnf-plugins'
+        end
+      end.compact
+    end
+
+    def protector_plugin_files
+      @protector_plugin_files ||= dnf_plugin_dirs.map do |d|
+        d + '/' + 'foreman-protector.py'
+      end
+    end
+
+    def install_version_locking(*)
+      protector_plugin_files.each do |f|
+        install_extras('foreman_protector/dnf/foreman-protector.py', f)
+      end
+      super
+    end
+
+    def install_extras(src, dest, override: false)
+      super
+    end
+
+    def version_locking_enabled?
+      File.exist?(PROTECTOR_CONFIG_FILE) && File.exist?(PROTECTOR_WHITELIST_FILE) &&
```

Comment:
hmm should I delegate this to parent class? However in parent class I would need to have if el7? condition for `PROTECTOR_PLUGIN_FILE` similar to `install_version_locking` method